datahub-web-react/src/app/entity/shared/containers/profile/header/PlatformContent/DatasetLink.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+import { Typography } from 'antd';
+import { EntityNameProperties, EntityType } from '../../../../../../../types.generated';
+import { ANTD_GRAY } from '../../../../constants';
+import { useEntityRegistry } from '../../../../../../useEntityRegistry';
+
+const DatasetText = styled(Typography.Text)`
+    font-size: 12px;
+    line-height: 20px;
+    color: ${ANTD_GRAY[7]};
+`;
+
+export const DatasetIcon = styled.span`
+    color: ${ANTD_GRAY[7]};
+    margin-right: 4px;
+    font-size: 12px;
+`;
+
+const StyledLink = styled(Link)`
+    white-space: nowrap;
+`;
+
+interface Props {
+    parentDatasetUrn: string | '';
+    parentDatasetProperties: EntityNameProperties;
+    parentDatasetIcon: JSX.Element;
+}
+
+function DatasetLink(props: Props) {
+    const { parentDatasetUrn, parentDatasetProperties, parentDatasetIcon } = props;
+    const entityRegistry = useEntityRegistry();
+    if (!parentDatasetUrn && !parentDatasetProperties) return null;
+
+    const datasetUrl = entityRegistry.getEntityUrl(EntityType.Dataset, parentDatasetUrn);
+    const datasetName = entityRegistry.getDisplayName(EntityType.Dataset, parentDatasetProperties);
```

Comment:
does this retrieve the correct dataset name? i believe `entityRegistry.getDisplayName` takes in a whole entity object, not just the properties object.

maybe it makes more sense to simply pass along the whole `parentDataset` object from `Preview` > `DefaultPreviewCard` > `DatasetLink` instead? then you can get the urn, the icon and the whatever else you may need where you need it

Reply:
Now we are passing `parentDataset` object from Preview.